test/puppetlabs/puppetdb/amq_migration_test.clj

```diff
@@ -52,6 +54,43 @@
     (mapv (comp str #(.toAbsolutePath %))
           (-> path-stream .iterator iterator-seq))))
 
+(defn processed-state []
+  (reduce (fn [acc metric]
+            (assoc acc
+                   (keyword metric)
+                   (->> ["global" metric]
+                        (meters/meter (get-in metrics-registries [:mq :registry]))
+                        meters/rates
+                        :total)))
+          {} ["processed" "retried" "discarded"]))
+
+(defn await-processed [await-how-many starting-processed-state timeout-ms]
```

Comment:
This looks good overall, but I was trying to see if I could think of another name... (await-commands 4 ...)?  Not sure - perhaps it's fine as is.